pkg/resourcegenerator/resourceutils/helpers.go

```diff
@@ -18,24 +21,42 @@ func ShouldScaleToZero(jsonReplicas *apiextensionsv1.JSON) bool {
 	return false
 }
 
+// MatchesRegex checks if a string matches a regexp
+func MatchesRegex(s string, pattern string) bool {
```

Comment:
```suggestion
func matchesRegex(s string, pattern string) bool {
```

Doesn't need to be exported